### DISCUSSION TITLE
Cache-control of SSG page

### DISCUSSION THREAD

**Discussion by rick-yo** [vercel/next.js#73520] [Help]
### Summary

For static files, like CSS or js
Nextjs set cache-control to `public, max-age=31536000, immutable`

but for the SSG page, for example, this one https://github.com/rick-yo/nextjs-demo/blob/main/pages/ssg.tsx
when SSG is fetched on client navigation, it fetch this json

`http://localhost:3000/_next/data/_t9KsXJPVdhHyQJCNIu-5/ssg.json`

Nextjs set cache-control of this json to `s-maxage=31536000`

why does it behave differently with static files? The JSON will never change

### Example

https://github.com/rick-yo/nextjs-demo/blob/main/pages/ssg.tsx

**Comment by denis23x** [comment]
Hi @rick-yo 

- Static Files (CSS/JS): These are typically pre-built, cached, and served directly from a Content Delivery Network (CDN) or server without modification. They rarely change per request.
- HTML: Even though HTML can be cached, it often needs to be generated dynamically to reflect the user’s current context or the latest state of the application.

In short, HTML files are not always treated as static because they usually need to incorporate dynamic, context-sensitive data, while CSS and JavaScript files remain unchanged regardless of the user or request.

**Comment by rick-yo** [comment]
Hi @denis23x 
Thanks for reply
but my question is about the `cache-control` difference of SSG JSON file and Static files
the SSG JSON file is generated on build time and never changes, it should be treated like Static files

**Reply by denis23x** [reply]
No

**Reply by rick-yo** [reply]
Hmm, why it doesn't make sense to you?
